src/SelectField/styles.js

```diff
@@ -329,3 +330,24 @@ export default css`
         transform: translate(8px, -26px) scale(0.75);
     }
 `
+
+export const arrowIcon = resolve`
+    svg {
+        fill: inherit;
+        height: 24px;
+        width: 24px;
+        vertical-align: middle;
+        pointer-events: none;
+    }
+`
+
+export const menuOverride = resolve`
+    max-height: 300px;
+    overflow-y: auto;
+`
+
+export const selectIconStyles = resolve`
```

Comment:
Do you need `resolve` here?

It is better if you can use `css`.

Reply:
The reason I used `resolve` is because I changed the status icons in `src/icons/Status.js` to reuse duplicate code, but styled-jsx prevents styled from bubbling down, so I couldn't just add a style for the svg in the `Valid` component as the svg will be rendered by the `Icon` component.

**I can revert that**, but then I think styled-jsx enforces an anti-pattern on us as I'm not able to create abstract components anymore unless I use inheritance.
The Icons all share this code:
```jsx
        <svg
            xmlns="http://www.w3.org/2000/svg"
            width="48"
            height="48"
            viewBox="0 0 48 48"
            className={[iconStyle.className, styles.className, className].join(
                ' '
            )}
        >
            {children}
            <style>{iconStyle.styles}</style>
            <style>{styles.styles}</style>
        </svg>
```

Reply:
I do not think that abstract Icon component is a helpful abstraction, it adds another layer of indirection. I prefer this style:

```js
const iconStyle = css`
 svg {
     width: 24px;
     height: 24px;
     margin-right: 8px;
}
`

export function Default({ className }) {
    return (
        <svg
            xmlns="http://www.w3.org/2000/svg"
            width="48"
            height="48"
            viewBox="0 0 48 48"
            className={className}
        >
            <path d="M0 0h48v48H0z" fill="none" />
            <style jsx>{`
                svg {
                    fill: ${colors.grey700};
                }
            `}</style>
            <style jsx>{iconStyle}</style>
        </svg>
    )
}

export function Valid({ className }) {
    return (
        <svg
            xmlns="http://www.w3.org/2000/svg"
            width="48"
            height="48"
            viewBox="0 0 48 48"
            className={className}
        >
            <path d="M24 4C12.95 4 4 12.95 4 24c0 11.04 8.95 20 20 20 11.04 0 20-8.96 20-20 0-11.05-8.96-20-20-20zm-4 30L10 24l2.83-2.83L20 28.34l15.17-15.17L38 16 20 34z" />
            <style jsx>{`
                svg {
                    fill: ${colors.blue600};
                }
            `}</style>
            <style jsx>{iconStyle}</style>
        </svg>
    )
}

export function Warning({ className }) {
    return (
        <svg
            xmlns="http://www.w3.org/2000/svg"
            width="48"
            height="48"
            viewBox="0 0 48 48"
            className={className}
        >
            <path d="M2 42h44L24 4 2 42zm24-6h-4v-4h4v4zm0-8h-4v-8h4v8z" />

            <style jsx>{`
                svg {
                    fill: ${colors.yellow500};
                }
            `}</style>
            <style jsx>{iconStyle}</style>
        </svg>
    )
}

export function Error({ className }) {
    return (
        <svg
            xmlns="http://www.w3.org/2000/svg"
            width="48"
            height="48"
            viewBox="0 0 48 48"
            className={className}
        >
            <path d="M24 4C12.96 4 4 12.95 4 24s8.96 20 20 20 20-8.95 20-20S35.04 4 24 4zm2 30h-4v-4h4v4zm0-8h-4V14h4v12z" />

            <style jsx>{`
                svg {
                    fill: ${colors.red500};
                }
            `}</style>
            <style jsx>{iconStyle}</style>
        </svg>
    )
}
```

The `colors.red500` could be a `theme.error` instead, and same for the other states. Simpler is better.

Reply:
>I can revert that, but then I think styled-jsx enforces an anti-pattern on us as I'm not able to create abstract components anymore unless I use inheritance.

I'm not so sure that I consider that an anti-pattern.

Reply:
> Could you help me understand why this is a useful abstraction?

For the following two reasons:
1. It's much easier to add more icons
1. It's very easy to change icon details

For me, it's harder to see the differences between the different icons if they're "cluttered" with the svg code, which my brain will always read first as I don't know whether there are differences until I read the whole code.

But I get the point that my way of thought/approaching code is not how other devs do it, so I totally understand why it's a cognitive overhead for the others. I'm curious how others see this!

Reply:
If the other icons do not share the same `viewBox` and/or dimensions? Then the those attributes would have to be parameters as well?

Now the icon abstraction would look kind of similar to the `svg`:

```
const Icon = ({ viewBox, height, width }) => <svg viewBox={viewBox} height={height} width={width}>{children}</svg>
```

Another concern I have is that the `path` element only makes sense inside of the XML namespace  of http://www.w3.org/2000/svg; it belongs as close to the `<svg>` tag as possible. 

🤔

Reply:
> If the other icons do not share the same viewBox and/or dimensions? 

This would be a good reason for refactoring of course, but that'd be changing requirements. From what I can tell so far, this is very unlikely to happen, so we'd be preparing for an unlikely eventuality.

But I'll revert the change. It's too unimportant to have a controversy about this

Reply:
See latest commit